kedro/contrib/io/matplotlib/matplotlib_s3_writer.py

```diff
@@ -0,0 +1,157 @@
+        >>> list_plot_writer.save(plots_list)
+
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        filepath: str,
+        bucket_name: str,
+        s3fs_args: Optional[Dict] = None,
+        credentials: Optional[Dict[str, Any]] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``MatplotlibS3Writer``.
+
+        Args:
+            bucket_name: Name of the bucket without "s3://" prefix.
+            filepath: Key path to a matplot object file.
+            s3fs_args: Arguments for ``S3FileSystem``. See
+                https://s3fs.readthedocs.io/en/latest/api.html#s3fs.core.S3FileSystem
+            credentials: Argument for ``client_kwargs``. If needed ``aws_access_key_id``
+                and ``aws_secret_access_key`` are provided here.
+            save_args: Save args passed to `plt.savefig`. See
+                https://matplotlib.org/api/_as_gen/matplotlib.pyplot.savefig.html
+        """
+
+        _credentials = copy.deepcopy(credentials) or {}
+
+        self._s3fs_args = copy.deepcopy(s3fs_args) or {}
+
+        self._filepath = filepath
+        self._save_args = save_args if save_args else dict()
+        self._bucket_name = bucket_name
+
+        self._s3 = S3FileSystem(client_kwargs=_credentials, **self._s3fs_args)
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            bucket_name=self._bucket_name,
+            filepath=self._filepath,
+            s3fs_args=self._s3fs_args,
+            save_args=self._save_args,
+        )
+
+    def _load(self) -> None:
+        raise DataSetError(
+            "Loading not supported for `{}`".format(self.__class__.__name__)
+        )
+
+    def _save(self, data: Union[figure, List[figure], Dict[str, figure]]) -> None:
+        if isinstance(data, list):
+            for index, plot in enumerate(data):
+                full_key_path = "/".join(
+                    [self._bucket_name, self._filepath, (str(index) + ".png")]
+                )
+                self._save_to_s3(full_key_path=full_key_path, plot=plot)
+
+        elif isinstance(data, dict):
+            for plot_name, plot in data.items():
+                full_key_path = "/".join([self._bucket_name, self._filepath, plot_name])
+                self._save_to_s3(full_key_path=full_key_path, plot=plot)
+
+        else:
+            full_key_path = "/".join([self._bucket_name, self._filepath])
+            self._save_to_s3(full_key_path=full_key_path, plot=data)
+
+    def _save_to_s3(self, full_key_path: str, plot: figure):
+        bytes_buffer = io.BytesIO()
+        plot.savefig(bytes_buffer, **self._save_args)
+
+        with self._s3.open(full_key_path, mode="wb") as s3_file:
+            s3_file.write(bytes_buffer.getvalue())
+
+    def _exists(self) -> bool:
+        load_path = "/".join([self._bucket_name, self._filepath])
+        return self._s3.isdir(load_path) or self._s3.isfile(load_path)
```

Comment:
`self._s3.isdir(load_path)` only checks the existence of the directory, but not checking if something is in the directory. You could use `walk` to check if there is anything in the path. https://s3fs.readthedocs.io/en/latest/api.html#s3fs.core.S3FileSystem.walk

Reply:
Ah did I not push something for this. I used `iterdir()`:

```
     def _exists(self) -> bool:
         return self._filepath.is_file() or (
-            self._filepath.is_dir() and bool(list(self._filepath.iterdir()))
+            self._filepath.is_dir() and bool(list(self._filepath.walk()))
         )

```

Reply:
Anyway, both seem to work!